src/lib/storage/table.cpp

```diff
@@ -18,75 +18,68 @@
 
 namespace opossum {
 
-Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
-}
+Table::Table(const ChunkOffset target_chunk_size) : _target_chunk_size{target_chunk_size} { create_new_chunk(); }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+  for (const auto& chunk : _chunks) {
+    resolve_data_type(type, [&](const auto data_type_t) {
+      using ColumnDataType = typename decltype(data_type_t)::type;
+      chunk->add_segment(std::make_shared<ValueSegment<ColumnDataType>>());
```

Comment:
Chunks are supposed to immutable. Adding new columns to locked chunks might lead to conflicts.